modules/javafx.graphics/src/main/java/javafx/scene/text/Text.java

```diff
@@ -1108,6 +1112,18 @@ public final PathElement[] underlineShape(int start, int end) {
         return getRange(start, end, TextLayout.TYPE_UNDERLINE);
     }
 
+    /**
+     * Returns the shape for the strike-through in local coordinates.
+     *
+     * @param start the beginning character index for the range
+     * @param end the end character index (non-inclusive) for the range
+     * @return an array of {@code PathElement} which can be used to create a {@code Shape}
+     * @since 25
+     */
+    public final PathElement[] getStrikeThroughShape(int start, int end) {
```

Comment:
I just noticed that this is named `getStrikeThroughShape` rather than `strikeThroughShape`. The latter would match the similar `rangeShape`, `caretShape`, and `underlineShape` methods. We had discussed that you might propose to fix [JDK-8341438](https://bugs.openjdk.org/browse/JDK-8341438) by deprecating `rangeShape` and `underlineShape` and adding new `getRangeShape` and `getUnderlineShape` methods (although that might be problematic given the `caretShape` property, which necessarily already has a no-arg `getCaretShape()` method).

Did you name this method `getStrikeThroughShape` with the intention that it would have the fixed semantics of applying the insets from its inception? If so, then I recommend documenting that fact in this method.

Reply:
Yes, you are right - and I plan to address this in https://bugs.openjdk.org/browse/JDK-8341438 (as it will be a separate change).

Reply:
We discussed this with @kevinrushforth - I am going to remove the `getStrikeThroughShape()` method from Text/TextFlow in this PR, and add it under either a separate ticket, or as a part of https://bugs.openjdk.org/browse/JDK-8341438

Reply:
Created https://bugs.openjdk.org/browse/JDK-8357594 for `Text::strikeThroughShape` and `TextFlow::getStrikeThroughShape`.